v2/src/runner.py

```diff
@@ -83,11 +102,19 @@ def _create_commands(spec: _RetrySpec) -> Sequence[str]:
             output_name=str(
                 i
             ),  # Ensure the `robot` command does not overwrite previous runs
-            previous_output=previous_output,
+            previous_output=outputs[-1] if outputs else None,
             variablefile=variant.variablefile,
             argumentfile=variant.argumentfile,
             retry_strategy=spec.strategy,
         )
-        commands.append(_create_command(runner_cfg))
-        previous_output = runner_cfg.output()
-    return commands
+        variants.append(_create_variant_command(runner_cfg))
+        outputs.append(runner_cfg.output())
+
+    return _RetryCommands(
+        variants=variants,
+        merge=_create_merge_command(
+            python_executable=spec.python_executable,
+            variants_outputs=outputs,
+            final_output=spec.outputdir() / "merged.xml",
```

Comment:
I simply hard-coded this for now. We should probably make this configurable or at least store the name in a variable once it's clear how we post-process these files further.